include/seqan3/core/bit_manipulation.hpp

```diff
@@ -96,4 +104,47 @@ constexpr uint8_t bit_scan_reverse(unsigned_t n)
 #endif
 }
 
+/*!\brief Convert the byte encoding of integer values to little-endian byte order.
+ * \ingroup core
+ * \tparam type The type of the value to convert; must model std::Integral.
+ * \param  in   The input value to convert.
+ * \returns the converted value in little-endian byte-order.
+ *
+ * \details
+ *
+ * This function swaps the bytes if the host system uses big endian. In this case only 1, 2, 4, or 8 byte big
+ * integral types are allowed as input. On host systems with little endian this function is a no-op and returns the
+ * unchanged input value. Other systems with mixed endianness are not supported.
+ */
+template <std::Integral type>
+constexpr type to_little_endian(type const in) noexcept
+{
+    if constexpr (seqan3::detail::endian::native == seqan3::detail::endian::little)
+    {
+        return in;
+    }
+    else if constexpr (seqan3::detail::endian::native == seqan3::detail::endian::big)
+    {
+        static_assert(sizeof(type) <= 8,
+                      "Can only convert the byte encoding for integral numbers with a size of up to 8 bytes.");
+        static_assert(is_power_of_two(sizeof(type)),
+                      "Can only convert the byte encoding for integral numbers whose byte size is a power of two.");
+
+        if constexpr (sizeof(type) == 2)
+            return htole16(in);
+        else if constexpr (sizeof(type) == 4)
+            return htole32(in);
+        else if constexpr (sizeof(type) == 8)
+            return htole64(in);
+        else
+            return in;  // single byte.
+    }
+    else
+    {
+        static_assert(seqan3::detail::endian::native == seqan3::detail::endian::little ||
+                      seqan3::detail::endian::native == seqan3::detail::endian::big,
```

Comment:
can't you omit the `seqan3::detail::` since this is in detail?

Sorry, just saw this. You can merge after this is fixed.